### DISCUSSION TITLE
Python / notebook display of "EXPLAIN" and "EXPLAIN ANALYZE" seems buggy

### DISCUSSION THREAD

**Discussion by benjamin-du** [duckdb/duckdb#8579] [Q&A] [answered]
Hi ! 

I've been trying to use EXPLAIN and EXPLAIN ANALYZE as per the documentation.

Alas, the output seems truncated:

The following snippet:

`
import pandas as pd

import duckdb

students = pd.DataFrame({"sid": [1, 2], "name": ["mathew", "james"]})

exams = pd.DataFrame({"sid": [1, 2], "note": ["10", "15"]})

duckdb.query("""EXPLAIN SELECT name FROM students JOIN exams USING (sid) WHERE name LIKE 'Ma%';""")
`

produces incomplete result, both in a terminal:

![image](https://github.com/duckdb/duckdb-web/assets/32103306/217db1fe-5f88-480f-a292-6abedec09641)

... and in a notebook:

![image](https://github.com/duckdb/duckdb-web/assets/32103306/f7e5edf6-6979-4532-8732-0052bf8efc0b)

Could you please tell me how to get the full output using either plain Python or (better yet) a Jupyter notebook ?

**Comment by l1t1** [comment]
try to save the result and print it

```python

>>>> s='HELLO\nW'
>>>> s
'HELLO\nW'
>>>> print(s)
HELLO
W
```

**Comment by Tishj** [comment, marked as answer]
I'd recommend using `.explain('analyze')` instead

**Comment by benjamin-du** [comment]
Bonjour Tishj, l1t1,

thanks for your answers, I did use the following:
print(duckdb.sql("SELECT * FROM Orders").explain())

and the schema works like a charm:

![image](https://github.com/duckdb/duckdb/assets/32103306/42a77f15-871c-4a6a-a8dc-3ed2fee79117)

However, where in the documentation is this command .explain() listed ?

**Reply by Mause** [reply]
It's listed here: https://duckdb.org/docs/archive/0.8.1/api/python/reference/#duckdb.DuckDBPyRelation.explain

**Reply by Mause** [reply]
Although it looks like it's missing a doc string, oops
